### ISSUE TITLE
bug on highligthing with kaitai 

### ISSUE THREAD

**Issue by accountgit** [Mahlet-Inc/hobbits#93]
A file generated with the following command

     echo 2f fe b8 9b 22 00 00 00  00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00  00 00 a4 0f 08 00 1a 06 08 af d0 bc 82 06 | xxd -r -p > file.bin

opened with hobbits with the following ksy:

    meta:
      file-extension: bla
      id: servermod
      endian: le
      imports:
        - /serialization/google_protobuf
    seq:
      - id: records
        type: record
        repeat: eos
    types:
      main_header:
        seq:
          - id: ip1
            type: u1
          - id: ip2
            type: u1
          - id: ip3
            type: u1
          - id: ip4
            type: u1
          - id: body_len
            type: u4le
      record:
        seq:
          - id: len_record
            type: main_header
            doc: Size of whole record, including all headers, footers and data
          - id: ctype
            type: u1
            doc: ctype
          - id: flow
            type: u1
            doc: flow
          - id: cid
            type: u4le
            doc: cid
          - id: did
            type: u4le
            doc: did
          - id: seq1
            type: u8le
            doc: seq1
          - id: command
            type: u2le
            doc: command
          - id: protobuf
            size: len_record.body_len - 24
            type: google_protobuf

When you open the file with the struct in kaitai IDE works correctly, but opening in hobbits when you select the protobuf part (the last 10 bytes - 08 af d0 bc 82 06 ) it's highligthed the start of the file.

If you concatenate some times and select the protobuf of any of the registers always it's highligthed the starting bytes of the file.

The command to generate with 3 concats is:

    echo 2f fe b8 9b 22 00 00 00  00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00  00 00 a4 0f 08 00 1a 06 08 af d0 bc 82 06 | xxd -r -p > file.bin ; echo 2f fe b8 9b 22 00 00 00  00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00  00 00 a4 0f 08 00 1a 06 08 af d0 bc 82 06 | xxd -r -p >> file.bin ; echo 2f fe b8 9b 22 00 00 00  00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00  00 00 a4 0f 08 00 1a 06 08 af d0 bc 82 06 | xxd -r -p >> file.bin

**Comment by hello-adam** [comment]
Thank you for reporting! I'll look into this soon. I know there are some issues with the kaitai parsing.

**Comment by hello-adam** [comment]
I can't seem to figure this out. Sometimes the "_debug" metadata from the parse has an absolute offset, for example with the `records[1].len_record` section:
```
{ '_debug': defaultdict(<class 'dict'>,
                        { 'body_len': {'end': 46, 'start': 42},
                          'ip1': {'end': 39, 'start': 38},
                          'ip2': {'end': 40, 'start': 39},
                          'ip3': {'end': 41, 'start': 40},
                          'ip4': {'end': 42, 'start': 41}}),
  '_io': <kaitaistruct.KaitaiStream object at 0x7fff98348070>,
  '_parent': <servermod.Servermod.Record object at 0x7fff60bae8b0>,
  '_root': <servermod.Servermod object at 0x7fff98368df0>,
  'body_len': 34,
  'ip1': 47,
  'ip2': 254,
  'ip3': 184,
  'ip4': 155}
```

But for the records[1].protobuff section, everything resets to relative offsetting:
```
{ '_debug': defaultdict(<class 'dict'>,
                        { 'pairs': { 'arr': [ {'end': 2, 'start': 0},
                                              {'end': 10, 'start': 2}],
                                     'end': 10,
                                     'start': 0}}),
  '_io': <kaitaistruct.KaitaiStream object at 0x7fff60bae730>,
  '_parent': None,
  '_root': <google_protobuf.GoogleProtobuf object at 0x7fff60bae6d0>,
  'pairs': [ <google_protobuf.GoogleProtobuf.Pair object at 0x7fff60bae670>,
             <google_protobuf.GoogleProtobuf.Pair object at 0x7fff60bae580>]}
```

There might be a pattern to when it is relative (in which case I could get the absolute offset from the parent chunk), but I don't notice any such pattern.

I could just be missing something obvious in the way I'm parsing the kaitai output. I will ask the kaitai folks.

**Comment by hello-adam** [comment]
Alright, @KOLANICH helped me out with this one, and it should be fixed in 0.46.6

https://github.com/kaitai-io/kaitai_struct/issues/879

**Comment by hello-adam** [comment]
should be good in 0.46.7
